server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,853 @@
+                    self.output_format(f, verify_tardir, "res_invalid_tb_links")
+                if (
+                    verify_tardir
+                    in self.results_check["res_incorrect_tb_dir_links"]._dict
+                ):
+                    f.write("\tIncorrectly constructed tar ball links:\n")
+                    self.output_format(f, verify_tardir, "res_incorrect_tb_dir_links")
+                if (
+                    verify_tardir
+                    in self.results_check["res_invalid_tb_dir_links"]._dict
+                ):
+                    f.write("\tTar ball links to invalid incoming location:\n")
+                    self.output_format(f, verify_tardir, "res_invalid_tb_dir_links")
+                if verify_tardir in self.results_check["unused_prefix_files"]._dict:
+                    f.write("\tTar ball links with unused prefix files:\n")
+                    self.output_format(f, verify_tardir, "unused_prefix_files")
+                if verify_tardir in self.results_check["missing_prefix_files"]._dict:
+                    f.write("\tTar ball links with missing prefix files:\n")
+                    self.output_format(f, verify_tardir, "missing_prefix_files")
+                if verify_tardir in self.results_check["bad_prefix_files"]._dict:
+                    f.write("\tTar ball links with bad prefix files:\n")
+                    self.output_format(f, verify_tardir, "bad_prefix_files")
+                if verify_tardir in self.results_check["bad_prefixes"]._dict:
+                    f.write("\tTar ball links with bad prefixes:\n")
+                    self.output_format(f, verify_tardir, "bad_prefixes")
+                if verify_tardir in self.results_check["unexpected_user_links"]._dict:
+                    f.write("\tTar ball links not configured for this user:\n")
+                    self.output_format(f, verify_tardir, "unexpected_user_links")
+                if verify_tardir in self.results_check["wrong_user_links"]._dict:
+                    f.write("\tTar ball links for the wrong user:\n")
+                    self.output_format(f, verify_tardir, "wrong_user_links")
+
+        return 0
+
+    def output_format(self, f, verify_tardir, dir_val):
+        for val in sorted(self.results_check[dir_val]._dict[verify_tardir]):
+            f.write("\t\t{}\n".format(val))
+        return 0
+
+
+def verify_results(rhier, hier, user_arg=False):
+    if user_arg:
+        results_hierarchy = os.path.join(rhier.config.USERS, user_arg)
+    else:
+        results_hierarchy = hier.path
+
+    for controller in hier.controller_check["verifylist"]:
+        tarball_dirs, unpacking_tarball_dirs = (list() for i in range(2))
+        if not os.path.isdir(os.path.join(rhier.config.ARCHIVE, controller)):
+            continue
+        dirent_entries = list()
+        for root, dirs, files in os.walk(
+            os.path.join(results_hierarchy, controller), topdown=True
+        ):
+            for name in files:
```

Comment:
There *must* be a way to get this done without having to do it one item at a time.